### ISSUE TITLE
Test "IDs-Should-Be-Derived-From-ResourceIDs" is failed for custom query property of ‘Extension/Microsoft_OperationsManagementSuite_Workspace/PartType/LogsDashboardPart’

### ISSUE THREAD

**Issue by LeonB87** [Azure/arm-ttk#361]
I’m creating an Azure Dashboard through ARM. On one of the dashboard items, I’ve created a Query that creates an property called “cycleId”.

This causes ARM-TTK to fail on the “IDs-Should-Be-Derived-From-ResourceIDs”

Snippet from the ARM template: ( I removed some of the code that is long and not necessary)

```json
{
    "position": {
        "x": 8,
        "y": 4,
        "colSpan": 7,
        "rowSpan": 3
    },
    "metadata": {
        "inputs": [
          "removed inputs for brevity"
        ],
        "type": "Extension/Microsoft_OperationsManagementSuite_Workspace/PartType/LogsDashboardPart",
        "settings": {
            "content": {
                "GridColumnsWidth": {
                    "timestamp": "145px",
                    "message": "194px",
                    "customDimensions": "351px",
                    "Type": "172px",
                    "Details": "522px"
                    ** Failing on this CycleId **,
                    "cycleId": "76px",
                    "totalStates": "96px"
                },
                "Query": " *removed beginning part of Query for brevity* | Project timestamp, message, cycleId, totalStates, processed\n",
                "PartTitle": "Policy Processing runs",
                "PartSubTitle": "Cycles"
            }
        },
        "filters": {
            "MsPortalFx_TimeRange": {
                "model": {
                    "format": "local",
                    "granularity": "auto",
                    "relative": "24h"
                }
            }
        }
    }
}
```

The error when running ARM-TTK on the azuredeploy.json file

```text
Errors    : {Property: "cycleId" must use one of the following expressions for an resourceId property:
                        extensionResourceId,resourceId,subscriptionResourceId,tenantResourceId,if,parameters,reference,variables,subscription,guid}
Warnings  : {}
Output    : {}
AllOutput : Property: "cycleId" must use one of the following expressions for an resourceId property:
                        extensionResourceId,resourceId,subscriptionResourceId,tenantResourceId,if,parameters,reference,variables,subscription,guid
Passed    : False
Group     : deploymentTemplate
Name      : IDs Should Be Derived From ResourceIDs
Timespan  : 00:00:00.1678200
File      : {Name, Extension, Text, FullPath…}
```

**Comment by LeonB87** [comment]
Suggested Fix

Adding the following code as an exception in **IDs-Should-Be-Derived-From-ResourceIDs.test.ps1**
```powershell
    if ( $id.ParentObject.type -match '^Microsoft\.Portal\/dashboards$' ) {
        continue
    }
```
According to https://docs.microsoft.com/en-us/azure/templates/microsoft.portal/2019-01-01-preview/dashboards?tabs=json The type **Microsoft.Portal/dashboards** doesn't have an ID property. 

Alternatively, We could exclude **Extension/HubsExtension/PartType/MonitorChartPart**

```powershell
    if ( $id.ParentObject.type -match '^Extension\/HubsExtension\/PartType\/MonitorChartPart$' ) {
        continue
    }
```

**Comment by LeonB87** [comment]
Added PR with suggested solution

**Comment by StartAutomating** [comment]
Thanks for the PR.  We'll probably want to go up to the .metadata property so that no other metadata with *id properties causes an issue.

**Comment by bmoore-msft** [comment]
So be fixed by #364 and #362
